src/main/java/com/cazsius/solcarrot/command/FoodListCommand.java

```diff
@@ -71,12 +72,13 @@ static int syncFoodList(CommandContext<CommandSourceStack> context, Player targe
 		sendFeedback(context.getSource(), localizedComponent("sync.success"));
 		return Command.SINGLE_SUCCESS;
 	}
-	
-	static int clearFoodList(CommandContext<CommandSourceStack> context, Player target) {
+
+	static final DynamicCommandExceptionType ERROR_NO_PERMISSION = new DynamicCommandExceptionType(object -> (Component)object);
+	static int clearFoodList(CommandContext<CommandSourceStack> context, Player target) throws CommandSyntaxException {
 		boolean isOp = context.getSource().hasPermission(2);
 		boolean isTargetingSelf = isTargetingSelf(context, target);
 		if (!isOp && !isTargetingSelf)
-			throw new CommandRuntimeException(localizedComponent("no_permissions"));
+			throw ERROR_NO_PERMISSION.create(localizedComponent("no_permissions"));
```

Comment:
Would it make more sense to have no argument to `.create()` and instead have it produce the `localizedComponent("no_permissions")` directly? Would avoid the cast, but I'm not sure how this API was designed.